base_layer/core/src/transactions/key_manager/inner.rs

```diff
@@ -182,87 +190,118 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
-                                        debug!(target: LOG_TARGET, "result less than 33");
-                                    }
-
-                                    return PublicKey::from_canonical_bytes(&result.data()[1..33])
-                                        .map_err(|e| KeyManagerServiceError::LedgerError(e.to_string()));
-                                },
-                                Err(e) => return Err(KeyManagerServiceError::LedgerError(e.to_string())),
+                match &self.wallet_type {
+                    // If we have the unique case of being a ledger wallet, and the key is a Managed EphemeralNonce, or
+                    // SenderOffset than we fetch from the ledger, all other keys are fetched below.
+                    WalletType::Ledger(ledger) => match TransactionKeyManagerBranch::from_key(branch) {
+                        TransactionKeyManagerBranch::SenderOffsetLedger | TransactionKeyManagerBranch::RandomKey => {
+                            #[cfg(not(feature = "ledger"))]
+                            {
+                                Err(KeyManagerServiceError::LedgerError(
+                                    "Ledger is not supported".to_string(),
+                                ))
                             }
-                        }
-                    }
 
-                    if &TransactionKeyManagerBranch::DataEncryption.get_branch_key() == branch {
-                        let view_key = ledger
-                            .view_key
-                            .clone()
-                            .ok_or(KeyManagerServiceError::LedgerViewKeyInaccessible)?;
-                        return Ok(PublicKey::from_secret_key(&view_key));
-                    }
+                            #[cfg(feature = "ledger")]
+                            {
+                                let public_key = ledger_get_public_key(
+                                    ledger.account,
+                                    *index,
+                                    TransactionKeyManagerBranch::from_key(branch),
+                                )
+                                .map_err(|e| KeyManagerServiceError::LedgerError(e.to_string()))?;
+                                Ok(public_key)
+                            }
+                        },
+                        TransactionKeyManagerBranch::DataEncryption => {
+                            let view_key = ledger
+                                .view_key
+                                .clone()
+                                .ok_or(KeyManagerServiceError::LedgerViewKeyInaccessible)?;
+                            Ok(PublicKey::from_secret_key(&view_key))
+                        },
+                        _ => Err(KeyManagerServiceError::BranchNotSupported(format!(
+                            "{}, {}",
+                            branch, self.wallet_type
+                        ))),
+                    },
+                    _ => {
```

Comment:
this is not correct, we need to handle the provided wallet as well. But the old code was broken as well, so this is fine as is, we can fix it in a new pr. This pr is already long enough